### ISSUE TITLE
S/PDIF kernel module

### ISSUE THREAD

**Issue by mill1000** [mill1000/raspdif#5]
The end goal of this project would to be a kernel module for S/PDIF output. Something along the lines [bcm2835-i2s]( https://github.com/raspberrypi/linux/blob/rpi-4.19.y/sound/soc/bcm/bcm2835-i2s.c) but for S/PDIF

**Comment by herrernst** [comment]
That would be very cool if it could just be used like an ALSA device!

**Comment by mill1000** [comment]
Yeah! I've contemplated making it a kernel module that provides a `/dev/raspdif` endpoint which could be written to. It would be a step in the right direction.

In the meantime, you inspired me to whip up this ALSA configuration for raspdif. You can either use it in your `.asoundrc` or `/etc/asound.conf`. With this config any ALSA compatible player should work.

https://github.com/mill1000/raspdif/blob/d682654c1547f721683f3692216656b6f67c7f13/asound.conf#L1-L22

I also added the [systemd service](https://github.com/mill1000/raspdif/blob/d682654c1547f721683f3692216656b6f67c7f13/raspdif.service) I've been using.

I'll update the README when I get a chance.

**Comment by mill1000** [comment]
Closing this as a "won't do" for now.

**Comment by mill1000** [comment]
Found this project today: [kiffie/rpi-i2s-spdif](https://github.com/kiffie/rpi-i2s-spdif). Same ideas as raspdif but the creator knew enough about kernel modules to make it work.

Reopening this issue since the above project would be a good reference
